flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/EventTimeTrigger.java

```diff
@@ -36,12 +36,17 @@ private EventTimeTrigger() {}
 	@Override
 	public TriggerResult onElement(Object element, long timestamp, TimeWindow window, TriggerContext ctx) throws Exception {
 		ctx.registerEventTimeTimer(window.maxTimestamp());
-		return TriggerResult.CONTINUE;
+
+		return (timestamp < ctx.getCurrentWatermark()) ?
```

Comment:
This should be `window.maxTimestamp() <= ctx.getCurrentWatermark()`. If the tests pass with this, then they also need to be adapted to fail with the current code and then work with the new code.